WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainSanitizer.kt

```diff
@@ -0,0 +1,24 @@
+package org.wordpress.android.ui.sitecreation.domains
+
+import java.util.Locale
+import javax.inject.Inject
+
+class SiteCreationDomainSanitizer
+@Inject constructor() {
+    /**
+     * Sanitizes a query by taking the value before it's first period if it's present,
+     * removes it's scheme and finally, removes any characters that aren't alphanumeric.
+     */
+    fun sanitizeDomainQuery(query: String) =
+            query.run {
+                val periodIndex = indexOf(".")
+                if (periodIndex > 0) {
+                    substring(0, periodIndex)
+                } else {
+                    this
+                }
+            }.replace("http://", "")
+                    .replace("https://", "")
+                    .replace("[^a-zA-Z0-9]".toRegex(), "")
+                    .toLowerCase(Locale.getDefault())
```

Comment:
I'm wondering if we want to use the default locale. `toLowerCase` doesn't produce 1:1 mapping in some languages. Since this is a URL it might make sense to use Locale.ROOT. Wdyt?

Reply:
Yes, this make sense. It's safer to use this than default and it's more likely to have the desired behavior in different languages.

Reply:
Resolved here 9156018